test/ecto/repo_test.exs

```diff
@@ -656,7 +656,7 @@ defmodule Ecto.RepoTest do
       assert schema.__meta__.prefix == "private"
     end
 
-    test "insert, update and insert_or_update parent schema_prefix overrides children schema_prefix" do
+    test "insert, update and insert_or_update parent schema_prefix does not override children schema_prefix" do
```

Comment:
Do we have a test that the parent does override the child if the child has no prefix?

Reply:
I have added two additional test cases so we now cover